buildSrc/src/main/groovy/org/apache/beam/gradle/VendorJavaPlugin.groovy

```diff
@@ -0,0 +1,146 @@
+      // when attempting to resolve dependency issues.
+      project.apply plugin: "project-report"
+
+      project.dependencies { compile "${config.dependency}" }
+
+      project.shadowJar {
+        config.packagesToRelocate.each { srcNamespace ->
+          relocate(srcNamespace, "${config.intoPackage}.${srcNamespace}")
+        }
+        classifier = null
+        mergeServiceFiles()
+        zip64 true
+      }
+
+      project.task('validateVendoring', dependsOn: 'shadowJar') {
+        inputs.files project.configurations.shadow.artifacts.files
+        doLast {
+          project.configurations.shadow.artifacts.files.each {
+            FileTree exposedClasses = project.zipTree(it).matching {
+              include "**/*.class"
+              exclude "org/apache/beam/**"
+            }
+            if (exposedClasses.files) {
+              throw new GradleException("$it exposed classes outside of org.apache.beam namespace: ${exposedClasses.files}")
+            }
+          }
+        }
+      }
+
+      project.apply plugin: 'maven-publish'
+
+      project.publishing {
+        repositories {
+          maven {
+            name "testPublicationLocal"
+            url "file://${project.rootProject.projectDir}/testPublication/"
+          }
+          maven {
+            url(project.properties['distMgmtSnapshotsUrl'] ?: isRelease(project)
+                    ? 'https://repository.apache.org/service/local/staging/deploy/maven2'
+                    : 'https://repository.apache.org/content/repositories/snapshots')
+
+            // We attempt to find and load credentials from ~/.m2/settings.xml file that a user
+            // has configured with the Apache release and snapshot staging credentials.
+            // <settings>
+            //   <servers>
+            //     <server>
+            //       <id>apache.releases.https</id>
+            //       <username>USER_TOKEN</username>
+            //       <password>PASS_TOKEN</password>
+            //     </server>
+            //     <server>
+            //       <id>apache.snapshots.https</id>
+            //       <username>USER_TOKEN</username>
+            //       <password>PASS_TOKEN</password>
+            //     </server>
+            //   </servers>
+            // </settings>
+            def settingsXml = new File(System.getProperty('user.home'), '.m2/settings.xml')
+            if (settingsXml.exists()) {
+              def serverId = (project.properties['distMgmtServerId'] ?: isRelease(project)
+                      ? 'apache.releases.https' : 'apache.snapshots.https')
+              def m2SettingCreds = new XmlSlurper().parse(settingsXml).servers.server.find { server -> serverId.equals(server.id.text()) }
+              if (m2SettingCreds) {
+                credentials {
+                  username m2SettingCreds.username.text()
+                  password m2SettingCreds.password.text()
+                }
+              }
+            }
+          }
+        }
+
```

Comment:
Please make sure the properties and pom.xml are part of the artifacts, the relevant code segments from BeamModulePlugin are https://github.com/apache/beam/blob/97e914e0bfd6a3bd2f248dfff2e6390be63ce7ea/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L820 to https://github.com/apache/beam/blob/97e914e0bfd6a3bd2f248dfff2e6390be63ce7ea/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L854

Reply:
This has come up on the dev@ mailing list a couple of times which is why I mentioned it here. I believe JB and Romain had more details about this but am forwarding along that original sentiment.

Reply:
:beam-vendor-guava-20_0:generatePomFileForMavenJavaPublication is failing while publishing. I suppose we need to add the pom file.

Failing build https://scans.gradle.com/s/6qpqxcfcmbiqq/

Reply:
I fixed that issue in https://github.com/apache/beam/pull/7042

Reply:
Thanks, will wait for it to merge